Maple2.Server.Web/Endpoints/Upload.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Maple2.Model.Enum;
+using Maple2.PacketLib.Tools;
+using Maple2.Web.Constants;
+using Microsoft.AspNetCore.Http;
+using Serilog;
+
+namespace Maple2.Web.Endpoints;
+
+public static class UploadEndpoint {
+    public static async Task<IResult> Post(HttpRequest request) {
+        Stream bodyStream = request.Body;
+
+        MemoryStream memoryStream = await CopyStream(bodyStream);
+        if (memoryStream.Length == 0) {
+            return Results.BadRequest();
+        }
+
+        IByteReader packet = new ByteReader(memoryStream.ToArray());
+
+        packet.ReadInt();
+        var type = (UgcType) packet.ReadInt();
+        packet.ReadLong();
+        long characterId = packet.ReadLong();
+        long ugcUid = packet.ReadLong();
+        int id = packet.ReadInt(); // item id, guild id, others?
+        packet.ReadInt();
+        packet.ReadLong();
+
+        byte[]? fileBytes = packet.ReadBytes(packet.Available);
+
+        return type switch {
+            UgcType.ProfileAvatar => HandleProfileAvatar(fileBytes, characterId),
+            _ => HandleUnknownMode(type)
+        };
+    }
+
+    private static IResult HandleProfileAvatar(byte[] fileBytes, long characterId) {
+        string filePath = $"{Target.DataDir}/profiles/{characterId}/";
+        Directory.CreateDirectory(filePath);
+
+        string uniqueFileName = Guid.NewGuid().ToString();
+
+        // Deleting old files in the character folder
+        DirectoryInfo di = new(filePath);
```

Comment:
no short variable names like this unless it's an obvious acronym (e.g. id). Please use `directory` or something